netcore/System.Private.CoreLib/src/System/Reflection/CustomAttribute.cs

```diff
@@ -104,12 +104,12 @@ static object[] GetPseudoCustomAttributes (Type type)
 			return attrs;
 		}
 
-		internal static object[] GetCustomAttributesBase (ICustomAttributeProvider obj, Type attributeType, bool inheritedOnly)
+		internal static Attribute[] GetCustomAttributesBase (ICustomAttributeProvider obj, Type attributeType, bool inheritedOnly)
 		{
-			object[] attrs;
+			Attribute[] attrs;
 
 			if (IsUserCattrProvider (obj))
-				attrs = obj.GetCustomAttributes (attributeType, true);
+				attrs = (Attribute[]) obj.GetCustomAttributes (attributeType, true);
```

Comment:
This doesn't seem right.  `obj` here is some user-written code, so it could be returning an `object[]` legitimately. I think you have to like... try to cast and if it fails go through one by one and copy the elements...

Reply:
I think CoreCLR just casts for this as well? https://github.com/dotnet/runtime/blob/master/src/coreclr/src/System.Private.CoreLib/src/System/Attribute.CoreCLR.cs#L23

Reply:
Yeah the docs explicitly say it's an array of objects representing attributes, so it doesn't have to be attribute[]

Reply:
I guess later on in the file we do the copy to a list individually, casting each time dance, so maybe I do need to do it here as well :(
I'll put together some tests to see what CoreCLR does in practice and attempt to match them or open an issue.